publication-rest/src/test/java/no/unit/nva/publication/update/UpdatePublicationHandlerTest.java

```diff
@@ -865,6 +930,561 @@ void shouldRejectUpdateIfSettingInstanceTypeNotAllowingFilesOnPublicationContain
+
+        var userName = randomString();
+        var institutionId = RandomPersonServiceResponse.randomUri();
+
+        var publication = createAndPersistPublicationWithoutDoiAndWithResourceOwner(userName, institutionId);
+        publicationService.publishPublication(UserInstance.fromPublication(publication), publication.getIdentifier());
+
+        var inputStream = createUnpublishHandlerRequest(publication.getIdentifier(), userName, institutionId, USER);
+        updatePublicationHandler.handleRequest(inputStream, output, context);
+
+        var response = GatewayResponse.fromOutputStream(output, Void.class);
+        assertThat(response.getStatusCode(), Is.is(IsEqual.equalTo(SC_ACCEPTED)));
+    }
+
+    @Test
+    void shouldReturnBadRequestWhenUnpublishingNotPublishedPublication() throws IOException, ApiGatewayException {
+        var unpublishedPublication = createAndPersistPublicationWithoutDoi(false);
+        var publisherUri = unpublishedPublication.getPublisher().getId();
+
+        var inputStream = createUnpublishHandlerRequest(unpublishedPublication.getIdentifier(), randomString(),
+                                                        publisherUri,
+                                                        AccessRight.MANAGE_RESOURCES_ALL);
+        updatePublicationHandler.handleRequest(inputStream, output, context);
+
+        var response = GatewayResponse.fromOutputStream(output, Void.class);
+        assertThat(response.getStatusCode(), Is.is(IsEqual.equalTo(SC_BAD_REQUEST)));
+    }
+
+    @Test
+    void shouldReturnBadRequestWhenCommentIsMissingWhenUnpublishingPublication()
+        throws ApiGatewayException, IOException {
+
+        var userName = randomString();
+        var institutionId = RandomPersonServiceResponse.randomUri();
+
+        var publication = createAndPersistPublicationWithoutDoiAndWithResourceOwner(userName, institutionId);
+        publicationService.publishPublication(UserInstance.fromPublication(publication), publication.getIdentifier());
+
+        var unpublishRequest = new UnpublishPublicationRequest();
+        var request = new HandlerRequestBuilder<UnpublishPublicationRequest>(restApiMapper)
+                          .withUserName(userName)
+                          .withCurrentCustomer(institutionId)
+                          .withAccessRights(institutionId, USER)
+                          .withBody(unpublishRequest)
+                          .withPathParameters(
+                              Map.of(PUBLICATION_IDENTIFIER, publication.getIdentifier().toString()));
+
+        updatePublicationHandler.handleRequest(request.build(), output, context);
+        var gatewayResponse = toGatewayResponseProblem();
+
+        assertThat(gatewayResponse.getStatusCode(), Is.is(IsEqual.equalTo(SC_BAD_REQUEST)));
+        assertThat(getProblemDetail(gatewayResponse), containsString(UNPUBLISH_REQUEST_REQUIRES_A_COMMENT));
+    }
+
+    // TODO: Should this return 200 OK?
```

Comment:
If it isn't delete…I don't think so. Delete + not found could be 200, but this is not Delete